### ISSUE TITLE
OpenGL errors when changing light colour

### ISSUE THREAD

**Issue by konstin** [supertuxkart/stk-code#1857]
The following warning is flooding the terminal when I change the colour of a LightNode using setColor:
```
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glColor4f is deprecated and not available in preview contexts.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_ENUM error generated. Cannot enable <cap> in the current profile.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glBegin is deprecated and not available in preview contexts.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glVertex3d is deprecated and not available in preview contexts.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glVertex3d is deprecated and not available in preview contexts.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glVertex3d is deprecated and not available in preview contexts.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glVertex3d is deprecated and not available in preview contexts.
[warn   ] GLWrap: OpenGL debug callback - API
[warn   ] GLWrap:     Error type : ERROR
[warn   ] GLWrap:     Severity : HIGH
[warn   ] GLWrap:     Message : GL_INVALID_OPERATION error generated. Function glEnd is deprecated and not available in preview contexts.
```

Graphics info:
```
[info   ] IrrDriver: OpenGL version: 4.3
[info   ] IrrDriver: OpenGL vendor: NVIDIA Corporation
[info   ] IrrDriver: OpenGL renderer: GeForce GTX 750 Ti/PCIe/SSE2
[info   ] IrrDriver: OpenGL version string: 4.3.0 NVIDIA 346.22
[info   ] GLDriver: ARB Buffer Storage Present
[info   ] GLDriver: ARB Base Instance Present
[info   ] GLDriver: ARB Draw Indirect Present
[info   ] GLDriver: ARB Compute Shader Present
[info   ] GLDriver: ARB Texture Storage Present
[info   ] GLDriver: ARB Texture View Present
[info   ] GLDriver: ARB Bindless Texture Present
[info   ] GLDriver: ARB Image Load Store Present
[info   ] GLDriver: ARB Shader Atomic Counters Present
[info   ] GLDriver: ARB Shader Storage Buffer Object Present
[info   ] GLDriver: ARB Multi Draw Indirect Present
[info   ] GLDriver: EXT Texture Compression S3TC Present
[info   ] GLDriver: ARB Uniform Buffer Object Present
[info   ] GLDriver: ARB Geometry Shader 4 Present
```

Am I doing something wrong in my code or is this a bug in the renderer?

**Comment by vlj** [comment]
You may also use AMD's CodeXL and ask it to report OpenGL errors/debug output callback.
For some reasons it also report issues with glLinkProgram that are irrelevant but it also gives exact location of the function that triggers the error.

**Comment by konstin** [comment]
fixed in d84d6381eaa94c27388df66582946bb558a1d70d
